visual_behavior/ophys/response_analysis/cell_metrics.py

```diff
@@ -0,0 +1,557 @@
+    metrics_table = metrics_table.merge(mean_response, on='cell_specimen_id')
+    metrics_table = metrics_table.merge(image_selectivity_index, on='cell_specimen_id')
+    metrics_table = metrics_table.merge(lifetime_sparseness, on='cell_specimen_id')
+    metrics_table = metrics_table.merge(fraction_responsive_trials, on='cell_specimen_id')
+    metrics_table = metrics_table.merge(reliability, on='cell_specimen_id')
+    metrics_table = metrics_table.merge(running_modulation_index, on='cell_specimen_id')
+    if condition == 'changes':
+        metrics_table = metrics_table.merge(hit_miss_modulation_index, on='cell_specimen_id')
+    metrics_table['ophys_experiment_id'] = ophys_experiment_id
+    metrics_table['ophys_session_id'] = ophys_session_id
+
+    metrics_table = metrics_table.reset_index()
+    # metrics_table = metrics_table.melt(id_vars=['cell_specimen_id', 'ophys_experiment_id', 'ophys_session_id'])
+    metrics_table['condition'] = condition
+    metrics_table['session_subset'] = session_subset
+    metrics_table['stimuli'] = stimuli
+    metrics_table['events'] = use_events
+
+    metrics_table = metrics_table.rename(columns={'variable': 'metric'})
+
+    return metrics_table
+
+
+def get_metrics_df_filename(ophys_experiment_id, condition, stimuli, session_subset, use_events):
+    if use_events:
+        trace_type = 'events'
+    else:
+        trace_type = 'dFF'
+    filename = 'experiment_id_' + str(ophys_experiment_id) + '_' + condition + '_' + stimuli + '_' + session_subset + '_' + trace_type
+    return filename
+
+
+def get_metrics_df_filepath(ophys_experiment_id, condition, stimuli, session_subset, use_events):
+    import platform
+    if platform.system() == 'Linux':
+        save_dir = r'/allen/programs/braintv/workgroups/nc-ophys/visual_behavior/single_cell_metrics'
+    else:
+        save_dir = r'\\allen\programs\braintv\workgroups\nc-ophys\visual_behavior\single_cell_metrics'
+    filename = get_metrics_df_filename(ophys_experiment_id, condition, stimuli, session_subset, use_events)
+    filepath = os.path.join(save_dir, 'cell_metrics', filename + '.h5')
+    return filepath
+
+
+def get_cell_metrics_table_for_experiment(ophys_experiment_id, condition, stimulus, session_subset, use_events):
+    filepath = get_metrics_df_filepath(ophys_experiment_id, condition, stimulus,
+                                       session_subset, use_events)
+    metrics_table = pd.read_hdf(filepath, key='df')
+    return metrics_table
+
+
+def load_cell_metrics_table_for_experiments(ophys_experiment_ids, condition, stimulus, session_subset, use_events):
+    import visual_behavior.data_access.loading as loading
+    ophys_experiment_table = loading.get_filtered_ophys_experiment_table(release_data_only=True)
+    ophys_experiment_table = loading.add_superficial_deep_to_experiments_table(ophys_experiment_table)
+
+    problem_expts = []
+    metrics_table = pd.DataFrame()
+    if ophys_experiment_ids == 'all_experiments':
```

Comment:
This is causing a future warning:
![image](https://user-images.githubusercontent.com/19944442/117506419-69a02900-af3a-11eb-9092-2bf1cad997df.png)
We should check the type first to make sure it's a string. Only then compare it to 'all_experiments'